### ISSUE TITLE
make status bar red when can't connect to configured server (bitcoin core or electrum)

### ISSUE THREAD

**Issue by AndySchroder** [sparrowwallet/sparrow#821]
If the bitcoin core RPC or electrum server is not available, we have a grey status bar instead of a colored one (yellow, green, or blue on the lower right), but this is not a clear distinction from offline mode. If we expect the server to be connected, but it can't, it should be red in the lower right and not gray. Also, the "Connection failed" error goes away after a while. So, we can be left in a state where we are not sure if we are intentionally in offline mode, or if it is trying to connect, but it can't.

Also, I think under server preferences, we should have an option for no server connection ("Offline"), rather than just "Public Server", "Bitcoin Core", or "Private Electrum", that way one will not inadvertently go online to a public server if they didn't intend to by accidentally clicking the toggle button on the lower right of the status bar.

**Comment by craigraw** [comment]
> So, we can be left in a state where we are not sure if we are intentionally in offline mode, or if it is trying to connect, but it can't.

There is merit to the "red toggle on failure" suggestion, but I wanted to note that the toggle pulses when it is trying to connect.

> Also, I think under server preferences, we should have an option for no server connection

I understand the suggestion, but I think it adds too much UI heft and potential for user confusion fwiw. It's easy enough to select one of the non-public server options if this is a real concern.

**Comment by AndySchroder** [comment]
> It's easy enough to select one of the non-public server options if this is a real concern.

Would prefer this to be the default then if you select "Later or Offline Mode" during initial startup. Currently if I select that "Later or Offline Mode" button, the "Public Server" tab is selected under server preferences.

**Comment by AndySchroder** [comment]
Also, if I:
1. Start Sparrow fresh
2. Choose "Configure Server"
3. Choose Private Electrum
4. Leave the server URL blank
5. Click Close
6. Click the toggle in the lower right of the status bar

Then it makes the toggle in the lower right stay blue and gives no error messages. What is it doing in this scenario?

**Comment by craigraw** [comment]
> I don't see the toggle pulsing when it is trying to connect. What color is it supposed to pulse?

![pulsing](https://user-images.githubusercontent.com/862166/217757216-3bfd8f78-e67c-4edd-9734-11f68475d686.gif)


> Then it makes the toggle in the lower right stay blue and gives no error messages. What is it doing in this scenario?

Nothing. I agree this appeared confusing though, so the toggle button is now disabled when no server is configured. This eliminates the Introduction dialog issue as well.

> If I try to unset a previously defined private electrum or bitcoin core server, it still gives some error message saying "Connection failed ...." that comes and goes.

Clearing the the URL host field for either of these two server types will now disable the server toggle as well.

Changes in 545342df.

**Comment by craigraw** [comment]
The toggle will simply be disabled if the URL host is empty, so it is impossible to trigger it.

**Comment by AndySchroder** [comment]
Right, but there can still be a non emtpy URL that is bad. In that case, what happens?

**Comment by craigraw** [comment]
The connection will fail, either immediately or after trying for a little while depending on the error. The toggle will return to an offline state, and the status bar will display the reason for the failure.
